test/testcondition.cpp

```diff
@@ -655,6 +655,13 @@ class TestCondition : public TestFixture {
               "}");
         ASSERT_EQUALS("", errout.str());
 
+        //TRAC #7567: "(a | 7) > 6U" is always true for unsigned a
+        check("void f(unsigned int a) {\n"
+              "assert( (a | 0x7) > 6U );\n" // statement always true for unsigned a
```

Comment:
could you add testcases for < <= == != >= also. I would prefer separate testcases. maybe you can put each code sample on 1 line to make it more compact.

    check("void f(unsigned a) { assert((a|7)>6U); }");
    ASSERT...